trieste/acquisition/rule.py

```diff
@@ -195,18 +197,25 @@ def acquire(
         :param state: Unused.
         :return: The single (or batch of) points to query, and `None`.
         """
+        if self._acquisition_function is None:
+            self._acquisition_function = self._builder.prepare_acquisition_function(
+                datasets, models
+            )
+        else:
+            self._acquisition_function = self._builder.update_acquisition_function(
+                self._acquisition_function, datasets, models
+            )
 
-        acquisition_function = self._builder.prepare_acquisition_function(datasets, models)
-        points = self._optimizer(search_space, acquisition_function)
+        points = self._optimizer(search_space, self._acquisition_function)
 
         if isinstance(self._builder, GreedyAcquisitionFunctionBuilder):
             for _ in range(
                 self._num_query_points - 1
             ):  # greedily allocate remaining batch elements
-                greedy_acquisition_function = self._builder.prepare_acquisition_function(
-                    datasets, models, pending_points=points
+                self._acquisition_function = self._builder.update_acquisition_function(
```

Comment:
Having an acq as an attribute of rule will be helpful for debugging aswell, e.g. plotting acq functions evolving and the like